iam_groups_authn/sync.py

```diff
@@ -284,13 +284,17 @@ def get_users_to_add(iam_users, db_users):
     Args:
         iam_users: List of that group's IAM users. (e.g. ["user1", "user2", "user3"])
         db_users: List of an instance's database users. (e.g. ["db-user1", "db-user2"])
+        database_type: Enum specifying type of database.
 
     Returns:
         missing_db_users: Set of names of DB user's needing to be inserted into instance.
     """
-    missing_db_users = [
-        user for user in iam_users if mysql_username(user) not in db_users
-    ]
+    if database_type == DatabaseVersion.mysql:
+        missing_db_users = [
+            user for user in iam_users if mysql_username(user) not in db_users
```

Comment:
is `mysql_username` performing a sanitation here? Can we do this once somewhere rather than in a few different places?

Reply:
For mysql database usernames are truncated, IAM user "user@test.com" would be added as "user" into the database. For this reason, each time we are awaiting the `iam_users_future`, i am sanitizing the output for mysql type databases.